src/databricks/labs/dqx/profiler/profiler.py

```diff
@@ -443,7 +443,7 @@ def _do_cast(value: str | None, typ: T.DataType) -> Any | None:
         if not value:
             return None
         if isinstance(typ, T.IntegralType):
-            return int(value)
+            return int(float(value))
```

Comment:
Let's not introduce another hack.

I think the fix is required one level up. The issue occurs for "stddev" or "mean" metrics. I think they should always be float. Therefore, I don't understand the current checking of the type in the `_process_metric` method:
```
if (typ in {T.IntegerType(), T.LongType()}) and metric in {"stddev", "mean"}:
```

I think it should be:
```
if metric in {"stddev", "mean"}:
```

This should solve the issue.